### PR TITLE
Add postgresql tests

### PR DESCRIPTION
**Pull request by dcermak**
Depends on https://github.com/SUSE/BCI-dockerfile-generator/pull/278 being deployed. Before that the postgres tests will fail.

### CODE REVIEW

**Comment by mloviska** [comment]
Shouldn't we try to use bind mount/volume with `postgresql` containers? Other than that, LGTM

**Comment by dcermak** [comment]
> Shouldn't we try to use bind mount/volume with `postgresql` containers? Other than that, LGTM

That is already happening automagically. The postgres container has a `VOLUME /var/lib/postgresql/data` directive, so a volume will be created by podman/docker automatically.

**Review by mloviska — DISMISSED** [review]

**Review by jlausuch — APPROVED** [review]
Neat! 
Do you think it would be worth adding more coverage like more crud operations with users, permissions, (besides tables and data)?

**Comment by dcermak** [comment]
> Neat! Do you think it would be worth adding more coverage like more crud operations with users, permissions, (besides tables and data)?

No, I think it would be far more important to test that all the environment variables used in the docker-entrypoint do "the right thing^TM"

**Comment by dcermak** [comment]
I have added some additional tests, please take a look. They will be failing currently though, because the necessary fixes are not yet deployed.

**Comment by dirkmueller** [comment]
@dcermak needs rebase on top of the php tests

File: tests/test_postgres.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            (None, _postgres_user),
+            (None, _other_pg_user),
+            (POSTGRES_PASSWORD, _other_pg_pw),
+            (None, _pgdata),
+        ):
+            env = {"POSTGRES_PASSWORD": pw}
+            if pg_user:
+                env["POSTGRES_USER"] = pg_user
+            if pgdata:
+                env["PGDATA"] = pgdata
```
Comment (dirkmueller) [inline tests/test_postgres.py:59]: why is there no "VOLUME {pgdata}" added to containerfile in this case? what is creating the volume here?

**Comment by dcermak** [comment]
@dirkmueller could you help me review the postgres scenario test matrix? There's still plenty of failures but I lost track whether these are supported or unsupported scenarios…

**Review by dirkmueller — APPROVED** [review]

**Review by dirkmueller — APPROVED** [review]

**Pull request merged**